parser/src/grammar.lalrpop

```diff
@@ -0,0 +1,144 @@
+use crate::{
+    ast::{
+        Boundary, BoundaryConstraints, Constraint, Expr, AirDef,
+        Identifier, Source, SourceSection, TraceCols, TraceColsGrp,
+        TraceColsGrpType, TransitionConstraints
+    }, error::{Error, ParseError::InvalidInt}, lexer::Tok
+};
+use std::str::FromStr;
+use lalrpop_util::ParseError;
+
+grammar;
+
+// AST NODE
+// ------------------------------------------------------------------------------------------------
```

Comment:
nit: I would use the "======" for top-level separators as we do in the Miden repo